app/src/main/java/org/fossasia/pslab/sensorfragment/SensorFragmentMain.java

```diff
@@ -55,37 +53,18 @@ public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         scienceLab = ScienceLabCommon.scienceLab;
 
-        if (scienceLab.isConnected()) {
-            i2c = scienceLab.i2c;
-            sensorAddr.put(0x60, "MCP4728");
-            sensorAddr.put(0x48, "ADS1115");
-            sensorAddr.put(0x23, "BH1750");
-            sensorAddr.put(0x77, "BMP180");
-            sensorAddr.put(0x5A, "MLX90614");
-            sensorAddr.put(0x1E, "HMC5883L");
-            sensorAddr.put(0x68, "MPU6050");
-            sensorAddr.put(0x40, "SHT21");
-            sensorAddr.put(0x39, "TSL2561");
-
-            try {
-                data = i2c.scan(null);
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
-            if (data != null) {
-                for (Integer myInt : data) {
-                    if (sensorAddr.get(myInt) != null) {
-                        dataAddress.add(String.valueOf(myInt));
-                        dataName.add(sensorAddr.get(myInt));
-                    }
-                }
-                for (String s : dataAddress) {
-                    tvData += s + ":" + sensorAddr.get(Integer.parseInt(s)) + "\n";
-                }
-            }
-            String[] dataDisp = dataName.toArray(new String[dataName.size()]);
-            adapter = new ArrayAdapter<>(getContext(), R.layout.sensor_list_item, R.id.tv_sensor_list_item, dataDisp);
-        }
+        i2c = scienceLab.i2c;
```

Comment:
Hello @hardik124 ! Seems like a very nice fix! Thanks for the PR! :) Wouldn't this line throw any error if the device is not connected?